src/azure-cli/azure/cli/command_modules/appservice/_validators.py

```diff
@@ -499,3 +507,25 @@ def validate_centauri_delete_function(cmd, namespace):
         raise ValidationError(
             "Invalid Operation. This function is currently present in your image",
             "Please modify your image to remove the function and provide an updated image.")
+
+
+def validate_registry_server(namespace):
+    if namespace.environment and namespace.registry_server:
+        if not namespace.registry_username or not namespace.registry_password:
+            if ACR_IMAGE_SUFFIX not in namespace.registry_server:
+                raise ValidationError("Usage error: --registry-server, --registry-password and"
+                                      " --registry-username are required together if not using Azure Container Registry")  # pylint: disable=line-too-long
+
+
+def validate_registry_user(namespace):
+    if namespace.environment and namespace.registry_username:
+        if not namespace.registry_server or (not namespace.registry_password and ACR_IMAGE_SUFFIX not in namespace.registry_server):  # pylint: disable=line-too-long
+            raise ValidationError("Usage error: --registry-server, --registry-password and"
+                                  " --registry-username are required together if not using Azure Container Registry")
+
+
+def validate_registry_pass(namespace):
+    if namespace.environment and namespace.registry_password:
+        if not namespace.registry_server or (not namespace.registry_username and ACR_IMAGE_SUFFIX not in namespace.registry_server):  # pylint: disable=line-too-long
+            raise ValidationError("Usage error: --registry-server, --registry-password and"
+                                  " --registry-username are required together if not using Azure Container Registry")
```

Comment:
Please use specific error type `RequiredArgumentMissingError` instead of `ValidationError`
```suggestion
                raise RequiredArgumentMissingError("Usage error: --registry-server, --registry-password and"
                                      " --registry-username are required together if not using Azure Container Registry")  # pylint: disable=line-too-long


def validate_registry_user(namespace):
    if namespace.environment and namespace.registry_username:
        if not namespace.registry_server or (not namespace.registry_password and ACR_IMAGE_SUFFIX not in namespace.registry_server):  # pylint: disable=line-too-long
            raise RequiredArgumentMissingError("Usage error: --registry-server, --registry-password and"
                                  " --registry-username are required together if not using Azure Container Registry")


def validate_registry_pass(namespace):
    if namespace.environment and namespace.registry_password:
        if not namespace.registry_server or (not namespace.registry_username and ACR_IMAGE_SUFFIX not in namespace.registry_server):  # pylint: disable=line-too-long
            raise RequiredArgumentMissingError("Usage error: --registry-server, --registry-password and"
                                  " --registry-username are required together if not using Azure Container Registry")
```